includes/Modules/Analytics.php

```diff
@@ -962,6 +962,33 @@ function ( $dimension_def ) {
 					}
 				}
 
+				$dimension_filters          = $data['dimensionFilters'];
+				$dimension_filter_instances = array();
+				if ( ! empty( $dimension_filters ) && ( is_string( $dimension_filters ) || is_array( $dimension_filters ) ) ) {
+					if ( is_string( $dimension_filters ) ) {
+						$dimension_filters = explode( ',', $dimension_filters );
+					} elseif ( is_array( $dimension_filters ) && ! wp_is_numeric_array( $dimension_filters ) ) { // If single object is passed.
+						$dimension_filters = array( $dimension_filters );
+					}
```

Comment:
@ivankruchkoff `$data['dimensionFilters']` should always be an associated array (see your `isValidDimensionFilters` function), so we don't need this checks. If the `dimensionFilters` is not an array, we can ignore it.

Reply:
Addressed.